code/game/machinery/_machines_base/machine_construction/_construction.dm

```diff
@@ -18,6 +18,9 @@
 
 // Return a change state define or a fail message to block transition.
 /obj/machinery/proc/cannot_transition_to(var/state_path, var/mob/user)
+	var/obj/item/stock_parts/network_lock/lock = get_component_of_type(/obj/item/stock_parts/network_lock)
+	if(istype(lock) && !allowed(user)) // Only check if we have a network_lock.
+		return MCS_BLOCK
```

Comment:
Should check that state_path is `deconstructed`; otherwise it will block every state transition.